lib/protobuf/dsl.ex

```diff
@@ -359,35 +381,39 @@ defmodule Protobuf.DSL do
     props
   end
 
-  defp generate_default_fields(syntax, msg_props) do
-    fields =
-      msg_props.field_props
-      |> Map.values()
-      |> Enum.reduce(%{}, fn props, acc ->
-        if props.oneof do
-          acc
-        else
-          Map.put(acc, props.name_atom, Protobuf.Builder.field_default(syntax, props))
-        end
-      end)
-
-    Enum.reduce(msg_props.oneof, fields, fn {key, _}, acc ->
-      Map.put(acc, key, nil)
-    end)
+  defp maybe_gen_defstruct(mod, message_props) do
+    if Module.defines?(mod, {:__struct__, 1}) do
+      IO.warn("""
+      Since v0.9.0 of the :protobuf library, structs are automatically generated for \
+      modules that call "use Protobuf". Remove the struct definition or regenerate the files \
+      with the latest version of the protoc-gen-elixir plugin. This warning \
+      will become an error in version 0.10.0+ of the :protobuf library.\
+      """)
+    else
+      gen_defstruct(message_props)
+    end
```

Comment:
Here we warn inside the gen function, but for `maybe_def_t_typespec` we have separate function for it. Should it be made consistent?

Reply:
I'm not sure that's needed since we only call the warning for `defstruct` once, but we need to call the warning for `t()` from two different places.